src/EncryptionOracle.sol

```diff
@@ -80,19 +86,24 @@ abstract contract EncryptionOracle is
         _;
     }
 
-    /// @notice Create a new oracle contract with a distributed public key
+    /// @notice Initialize a new oracle contract with a distributed public key
     /// @dev The distributed key is created by an on-chain DKG process
     /// @dev Verify the key by checking all DKG contracts deployed by Medusa operators
     /// @param _distKey An (x, y) point on an elliptic curve representing a public key previously created by medusa nodes
     /// @param _relayer that is trusted to deliver reencryption results
     /// @param _submissionFee for submitCiphertext()
     /// @param _reencryptionFee for requestReencryption()
-    constructor(
+    function _initialize(
         G1Point memory _distKey,
+        address _owner,
         address _relayer,
         uint96 _submissionFee,
         uint96 _reencryptionFee
-    ) ThresholdNetwork(_distKey) {
+    ) internal onlyInitializing {
+        Ownable._initializeOwner(_owner);
```

Comment:
Isn't the owner set when we deploy it the first time (Even though unitialized) ?

Reply:
with openzeppelin/Ownable this is true.

I changed to solady/Ownable because I want to explicitly set the owner. This is because the OracleFactory deploys this contract, but I want the EOA/deployer account to be the owner.

Reply:
Got it thx